target/linux/ipq806x/files-6.6/arch/arm/boot/dts/qcom/qcom-ipq8064-e8350-v1.dts

```diff
@@ -0,0 +1,428 @@
+					reg = <0x0190000 0x0050000>;
+					read-only;
+				};
+				
+				partition@1e0000 {
+					label = "0:oldnss1";
+					reg = <0x01e0000 0x0020000>;
+					read-only;
+				};
+				
+				partition@200000 {
+					label = "0:nvram";
+					reg = <0x0200000 0x0020000>;
+					read-only;
+				};
+				
+				partition@220000 {
+					label = "0:oldkernel";
+					reg = <0x0220000 0x01e0000>;
+					read-only;
+				};
+			};
+		};
+	};
+};
+
+&hs_phy_0 {
+	status = "okay";
+};
+
+&ss_phy_0 {
+	status = "okay";
+};
+
+&usb3_0 {
+	status = "okay";
+};
+
+&hs_phy_1 {
+	status = "okay";
+};
+
+&ss_phy_1 {
+	status = "okay";
+};
+
+&usb3_1 {
+	status = "okay";
+};
+
+&pcie0 {
+	status = "okay";
+	
+	max-link-speed = <1>;
+};
+
+&pcie1 {
+	status = "okay";
+};
+
+&mdio0 {
+	status = "okay";
+	
+	pinctrl-0 = <&mdio0_pins>;
+	pinctrl-names = "default";
+	
+	/* Switch from documentation require at least 12ms for reset */
+	reset-gpios = <&qcom_pinmux 63 GPIO_ACTIVE_HIGH>;
+	reset-post-delay-us = <12000>;
+	
+	switch@10 {
+		compatible = "qca,qca8337";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		reg = <0x10>;
+		
+		ports {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			
+			port@0 {
+				reg = <0>;
+				label = "cpu";
+				ethernet = <&gmac1>;
+				phy-mode = "rgmii";
+				tx-internal-delay-ps = <1000>;
+				rx-internal-delay-ps = <1000>;
+				
+				fixed-link {
+					speed = <1000>;
+					full-duplex;
+				};
+			};
+			
+			port@1 {
+				reg = <1>;
+				label = "lan1";
+				phy-mode = "internal";
+				phy-handle = <&phy_port1>;
+			};
+			
+			port@2 {
+				reg = <2>;
+				label = "lan2";
+				phy-mode = "internal";
+				phy-handle = <&phy_port2>;
+			};
+			
+			port@3 {
+				reg = <3>;
+				label = "lan3";
+				phy-mode = "internal";
+				phy-handle = <&phy_port3>;
+			};
+			
+			port@4 {
+				reg = <4>;
+				label = "lan4";
+				phy-mode = "internal";
+				phy-handle = <&phy_port4>;
+			};
+			
+			port@5 {
+				reg = <5>;
+				label = "wan";
+				phy-mode = "internal";
+				phy-handle = <&phy_port5>;
+			};
+			
+			port@6 {
+				reg = <6>;
+				label = "cpu";
+				ethernet = <&gmac2>;
+				phy-mode = "sgmii";
+				qca,sgmii-enable-pll;
+				
+				fixed-link {
+					speed = <1000>;
+					full-duplex;
+				};
+			};
+		};
+		
+		mdio {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			
+			phy_port1: phy@0 {
+				reg = <0>;
+			};
+			
+			phy_port2: phy@1 {
+				reg = <1>;
+			};
+			
+			phy_port3: phy@2 {
+				reg = <2>;
+			};
+			
+			phy_port4: phy@3 {
+				reg = <3>;
+			};
+			
+			phy_port5: phy@4 {
+				reg = <4>;
+			};
+		};
+	};
+};
+
+&gmac1 {
+	status = "okay";
+	
+	phy-mode = "rgmii";
+	qcom,id = <1>;
+	
+	pinctrl-0 = <&rgmii2_pins>;
+	pinctrl-names = "default";
+	
+	fixed-link {
+		speed = <1000>;
+		full-duplex;
+	};
+};
+
+&gmac2 {
+	status = "okay";
+	
+	phy-mode = "sgmii";
+	qcom,id = <2>;
+	
+	fixed-link {
+		speed = <1000>;
+		full-duplex;
+	};
+};
+
+&tcsr {
+	qcom,usb-ctrl-select = <TCSR_USB_SELECT_USB3_DUAL>;

```

Comment:
This is a good time to ask for this... Is it actually required? Does usb works without this?

Reply:
I've tried completely commenting it out, and the router stuck in boot (waited 5 minutes). After reintroducing it went back to a normal operation. So, for this model it seems to be required.

Reply:
Mhhh no serial? Sorry if i'm asking this but I always wanted to upstream this tcsr thing and wa always confused by it.

Reply:
Up until now there was no need in serial (thanks to other members who did research earlier).
But sure, I can try this out with serial connected. Will take some time tho (but most probably until the end of the week there will be some logs).

Reply:
> Mhhh no serial? Sorry if i'm asking this but I always wanted to upstream this tcsr thing and wa always confused by it.

@Ansuel, actually I've managed to boot it, if completely commented out the whole section of tcsr.

```
[    6.087604] xhci-hcd xhci-hcd.0.auto: xHCI Host Controller
[    6.087654] xhci-hcd xhci-hcd.0.auto: new USB bus registered, assigned bus number 1
[    6.092195] xhci-hcd xhci-hcd.0.auto: hcc params 0x0228f065 hci version 0x100 quirks 0x0000008002000010
[    6.099577] xhci-hcd xhci-hcd.0.auto: irq 47, io mem 0x10000000
[    6.109149] xhci-hcd xhci-hcd.0.auto: xHCI Host Controller
[    6.114892] xhci-hcd xhci-hcd.0.auto: new USB bus registered, assigned bus number 2
[    6.120375] xhci-hcd xhci-hcd.0.auto: Host supports USB 3.0 SuperSpeed
[    6.128566] hub 1-0:1.0: USB hub found
[    6.134603] hub 1-0:1.0: 1 port detected
[    6.138572] usb usb2: We don't know the algorithms for LPM for this host, disabling LPM.
[    6.142865] hub 2-0:1.0: USB hub found
[    6.150429] hub 2-0:1.0: 1 port detected
[    6.155681] xhci-hcd xhci-hcd.1.auto: xHCI Host Controller
[    6.158054] xhci-hcd xhci-hcd.1.auto: new USB bus registered, assigned bus number 3
[    6.163543] xhci-hcd xhci-hcd.1.auto: hcc params 0x0228f065 hci version 0x100 quirks 0x0000008002000010
[    6.170925] xhci-hcd xhci-hcd.1.auto: irq 48, io mem 0x11000000
[    6.180492] xhci-hcd xhci-hcd.1.auto: xHCI Host Controller
[    6.186244] xhci-hcd xhci-hcd.1.auto: new USB bus registered, assigned bus number 4
[    6.191806] xhci-hcd xhci-hcd.1.auto: Host supports USB 3.0 SuperSpeed
[    6.199734] hub 3-0:1.0: USB hub found
[    6.205951] hub 3-0:1.0: 1 port detected
[    6.209908] usb usb4: We don't know the algorithms for LPM for this host, disabling LPM.
[    6.214133] hub 4-0:1.0: USB hub found
[    6.221854] hub 4-0:1.0: 1 port detected
```
trying to attach a USB device fails to the first port
```
[  193.543764] usb usb4-port1: connect-debounce failed
```
works for a second one (e-sata):
```
[  246.511736] usb 1-1: new high-speed USB device number 2 using xhci-hcd
[  246.697096] usb-storage 1-1:1.0: USB Mass Storage device detected
[  246.697437] usb-storage 1-1:1.0: Quirks match for vid 174c pid 55aa: 400000
[  246.702390] scsi host0: usb-storage 1-1:1.0
[  247.773322] scsi 0:0:0:0: Direct-Access     SABRENT  ASM225           0    PQ: 0 ANSI: 6
[  247.774786] sd 0:0:0:0: [sda] 1875385008 512-byte logical blocks: (960 GB/894 GiB)
[  247.780964] sd 0:0:0:0: [sda] Write Protect is off
[  247.787939] sd 0:0:0:0: [sda] Mode Sense: 43 00 00 00
[  247.788431] sd 0:0:0:0: [sda] Write cache: enabled, read cache: enabled, doesn't support DPO or FUA
[  247.802895]  sda: sda1
[  247.803136] sd 0:0:0:0: [sda] Attached SCSI disk
```
Full log is here: https://microbin.fenriv.co.ua:444/raw/sloth-bird

Reply:
Really interesting... Can you contact you later for further testing about this? Thanks a lot for the tests.

Reply:
> Really interesting... Can you contact you later for further testing about this? Thanks a lot for the tests.

Yes, sure, you welcome 😊. 
I will mark then this note as resolved.